src/plugins/files/server/blob_storage_service/adapters/es/es.ts

```diff
@@ -65,23 +65,29 @@ export class ElasticsearchBlobStorageClient implements BlobStorageClient {
   }
 
   /**
-   * This function acts as a singleton i.t.o. execution: it can only be called once.
-   * Subsequent calls should not re-execute it.
-   *
-   * There is a known issue where calling this function simultaneously can result
-   * in a race condition where one of the calls will fail because the index is already
-   * being created. This is only an issue for the very first time the index is being
-   * created.
+   * This function acts as a singleton i.t.o. execution: it can only be called once per index.
+   * Subsequent calls for the same index should not re-execute it.
    */
-  private static createIndexIfNotExists = once(
-    async (index: string, esClient: ElasticsearchClient, logger: Logger): Promise<void> => {
+  protected static createIndexIfNotExists = memoize(
```

Comment:
I assume memoize works with the first argument by default?

Reply:
@Dosant - Correct 😄